src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -3873,8 +4580,19 @@ def spawn(self, launcher, cu):
 
             # The actual command line, constructed per launch-method
             try:
-                launch_command, hop_cmd = \
-                    launcher.construct_command(cu['description']['executable'],
+                if launcher.name == 'LaunchMethodYARN':
```

Comment:
We should  go through all parts of the code where 'YARN' is checked -- but this one specifically: the construct_command signature should be kept consistent.  Lets discuss this with Mark.